### PR TITLE
enable kube-state-metrics for SKE Usage

### PR DESCRIPTION
**Pull request by Duciwuci**
includes:
* kube_pod_created

**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

We want to provide the usability of the metrics in our dashboards / alerts and since the general allowlist are not including some of the metrics, we are enabling them manually.

### CODE REVIEW

**Comment by hown3d** [comment]
allowedMetrics list doesn't need to be adjusted. Our monex seed config scrapes everything from kube-state-metrics that is returned from it.

File: pkg/component/kubestatemetrics/monitoring.go [hunk 1]
```diff
@@ -122,6 +123,7 @@ var (
 		monitoringMetricKubePodInfo,
 		monitoringMetricKubePodLabels,
 		monitoringMetricKubePodOwner,
+		monitoringMetricKubePodCreated,
```
Comment (hown3d) [inline pkg/component/kubestatemetrics/monitoring.go:126]: allowedMetrics list doesn't need to be adjusted. Our monex seed config scrapes everything from kube-state-metrics that is returned from it.

**Comment by Duciwuci** [comment]
You have to remove the complete flag...

File: pkg/component/kubestatemetrics/monitoring_test.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ metric_relabel_configs:
...
...e_spec_unschedulable|kube_node_status_allocatable|kube_node_status_capacity|kube_node_status_condition|kube_persistentvolumeclaim_resource_requests_storage_bytes|kube_pod_container_info|kube_pod_container_resource_limits|kube_pod_container_resource_requests|kube_pod_container_status_restarts_total|kube_pod_info|kube_pod_labels|kube_pod_owner|kube_pod_created|kube_pod_spec_volumes_persistentvolumeclaims_info|kube_pod_status_phase|kube_pod_status_ready|kube_replicaset_owner|kube_statefulset_metadata_generation|kube_statefulset_replicas|kube_statefulset_status_observed_generation|kube_statefulset_status_replicas|kube_statefulset_status_replicas_current|kube_statefulset_status_replicas_ready|kube_statefulset_status_replicas_updated|kube_verticalpodautoscaler_status_recommendation_containerrecommendations_target|kube_verticalpodautoscaler_status_recommendation_containerrecommendations_upperbound|kube_verticalpodautoscaler_status_recommendation_containerrecommendations_lowerbound|kube_verticalpodautoscaler_spec_resourcepolicy_container_policies_minallowed|kube_verticalpodautoscaler_spec_resourcepolicy_container_policies_maxallowed|kube_verticalpodautoscaler_spec_updatepolicy_updatemode)$
```
Comment (hown3d) [inline pkg/component/kubestatemetrics/monitoring_test.go:109]: kube_pod_owner was missing in this list

**Comment by hown3d** [comment]
we don't need kube_pod_created anymore, closing.

**Pull request closed without merging**